test-crawler/crawler_test.go

```diff
@@ -24,4 +23,22 @@ func TestExtractPublicMethodsFromFile(t *testing.T) {
 	assert.Equal(t, "HelloEventWithParameter", fnsAnn[1].Name)
 	assert.Equal(t, "(param string)", fnsAnn[1].InputParams)
 	assert.Equal(t, "(string, error)", fnsAnn[1].ReturnValues)
+
+	assert.Equal(t, "FunctionWithoutParameters", fnsAnn[2].Name)
+	assert.Equal(t, "()", fnsAnn[2].InputParams)
+	assert.Equal(t, "", fnsAnn[2].ReturnValues)
+
+	assert.Equal(t, "FunctionWithPointerReturnValue", fnsAnn[3].Name)
+	assert.Equal(t, "()", fnsAnn[3].InputParams)
+	assert.Equal(t, "*Event", fnsAnn[3].ReturnValues)
+}
+
+func TestCrawlFolderForSystemMethods(t *testing.T) {
+	systemMethods, err := crawlFolderForSystemMethods("./mocks")
+	if err != nil {
+		t.Errorf("got error: %v", err.Error())
+	}
+
+	assert.Equal(t, systemMethods.System, "mocks")
+	assert.Equal(t, 6, len(systemMethods.Methods))
```

Comment:
This unit test actually tests 2 things:
- Crawling a folder for files
- Extracting methods from each file

While the behavior `crawlFolderForSystemMethods` is not very complex, we should add more unit tests that target its parts.

When writing a unit test, we want it to test a single behavior, if possible. Therefore, one type of tests should test the method `ListGoFilesInFolder`, and include several tests for as many branches in the function as possible.

Reply:
Since we have nested function calls, I will extract testing of `ListGoFilesInFolder` in it's package module. If that function is well tested, no need to test single `crawlFolderForSystemMethods` since it is only combined call of `ListGoFilesInFolder` and `crawlSingleFileForFunctions`

Reply:
Agreed. You can remove the test for `crawlFolderForSystemMethods`, and just write the tests for the two methods it encapsulates.
Of course, should this behavior change in the future, we will add new unit tests.